### ISSUE TITLE
Document `conda search --outdated`

### ISSUE THREAD

**Issue by maxkapur** [conda/conda#14418]
### Checklist

- [x] I added a descriptive title
- [x] I searched open reports and couldn't find a duplicate

### What happened?

I learned about the `conda search --outdated` command from [this StackOverflow post](https://stackoverflow.com/questions/24958833/bulk-package-updates-using-conda), but I see no mention of the `--outdated` flag in either the CLI help (`conda search --help`) or online documentation (https://docs.conda.io/projects/conda/en/latest/commands/search.html).

Is this feature still supported? Are there any additional options available (e.g. to filter for packages listed explicitly in `environment.yml` but ignore outdated dependencies)?

I would be happy to submit a documentation PR if you could point me in the right direction.

**Comment by travishathaway** [comment]
@maxkapur,

I looked at this today and discovered this option does not do anything. This is a feature that is no longer supported, so created a PR to document this feature would not make sense.

Feel free to submit a feature request if you would like, but I personally think that this would be a better option to add to the `conda list` command rather than the `conda search` command.

**Comment by travishathaway** [comment]
In to reduce future, confusion, we will be removing this option from `conda search`.

**Comment by maxkapur** [comment]
Thank you for the swift response. It seems like all that's required to eliminate the dead code is to remove the following argparse lines in `conda/cli/main_search.py`:

```python
    p.add_argument(
        "-o",
        "--outdated",
        action="store_true",
        help=SUPPRESS,
    )
```

Regarding a "list outdated packages" feature: I found that `conda update --all --dry-run` was adequate for my use case—namely, to identify if any packages are being held back because of the versions specified in `environment.yml`. (Observe that this way, you only see updates that are actually achievable given dependency constraints.) In my CI job, I implemented it like this:

```shell
# Immediately after conda create, conda activate
conda update --all --dry-run | tee grep -q "requested packages already installed"
```

which exits with 0 if the phrase is found (OK) or 1 if not (outdated packages exist).

However, I realize that the exact wording "requested packages already installed" could probably change between non-breaking releases.

What do you think about adding an `"already_installed": True/False` field to the `--json` output here in the `handle_txn()` function of `conda/cli/install.py` (called by `conda update`)?

```
                common.stdout_json_success(
                    message="All requested packages already installed."
                )
```

Happy to issue a PR or feature request if you'd like.

**Comment by travishathaway** [comment]
@maxkapur,

Yes, I agree that your solution using grep isn't the most robust.

Before I give you advice, I want to let you know that what we really need for the JSON output of conda commands are clearly defined and versioned schemas giving users an exact definition of the all the fields and what they can be used for. Unfortunately, we do not currently have that.

With that being said, I think the best way to check the successful or "no operation" run of `conda update --all` would be to check the JSON output and see the presence of `success: true` and the absence of `actions`. That's not particular satisfying, but I think it's better than relying on the `message` field, which as you implied, may be liable to change in the future.

For example, this is what the JSON output looks like for `conda update --all` when no packages can be updated:

```json
{
  "message": "All requested packages already installed.",
  "success": true
}
```

This is a snippet of what it looks like when packages can be updated:

```json
{
  "actions": {
    "FETCH": [
    ],
  },
  "dry_run": true,
  "prefix": "/home/user/opt/conda",
  "success": true
}
```

Hope this helps. I don't believe it's necessary to add a new `already_installed` field considering what I've said above.

**Comment by maxkapur** [comment]
Thanks for the tip! I agree, leveraging the existing API is a better solution.
